ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py

```diff
@@ -39,44 +39,66 @@ def test_definition(self) -> dict | None:
-
-                if ethernet_interface["type"] == "routed":
-                    src_ip = str(ip_interface(ethernet_interface["ip_address"]).ip)
-                    dst_ip = str(ip_interface(peer_interface_ip).ip)
-                    custom_field = (
-                        f"Source: {self.device_name}_{ethernet_interface['name']} - Destination:"
-                        f" {ethernet_interface['peer']}_{ethernet_interface['peer_interface']}"
+            if ethernet_interface["type"] == "routed":
+                try:
+                    ethernet_interface_ip = get(ethernet_interface, "ip_address", required=True)
+                except AristaAvdMissingVariableError as e:
+                    LOGGER.warning("Ethernet interface '%s' is missing the variable '%s'.", ethernet_interface["name"], str(e))
+                    continue
+
+                if ethernet_interface["shutdown"]:
+                    LOGGER.info("Ethernet interface '%s' is shutdown. 'VerifyReachability' is skipped for this interface.", ethernet_interface["name"])
+                    continue
+
+                if (peer := ethernet_interface["peer"]) not in self.hostvars or not get(self.hostvars[peer], "is_deployed", default=True):
+                    LOGGER.info(
+                        "Peer '%s' is not configured by AVD or is marked as not deployed. 'VerifyReachability' from interface '%s' to this peer is skipped.",
+                        peer,
+                        ethernet_interface["name"],
                     )
-                    anta_tests.append(
-                        {
-                            "VerifyReachability": {
-                                "hosts": [{"source": src_ip, "destination": dst_ip, "vrf": "default", "repeat": 1}],
-                                "result_overwrite": {"categories": self.categories, "description": self.description, "custom_field": custom_field},
-                            }
-                        }
+                    continue
+
+                try:
+                    peer_ethernet_interfaces = get(self.hostvars[peer], "ethernet_interfaces", required=True)
+                    peer_ethernet_interface = get_item(
+                        peer_ethernet_interfaces,
+                        "name",
+                        ethernet_interface["peer_interface"],
+                        required=True,
+                    )
+                    peer_ethernet_interface_ip = get(peer_ethernet_interface, "ip_address", required=True)
+                except AristaAvdMissingVariableError:
+                    LOGGER.warning(
+                        "Ethernet interface '%s' connected peer '%s' is missing interface '%s' or his variable 'ip_address' is missing.",
+                        ethernet_interface["name"],
+                        peer,
+                        ethernet_interface["peer_interface"],
                     )
+                    continue
```

Comment:
Maybe write a helper called `_get_peer_interface_ip` that handles the error and returns the ip_address or None and you can then do the continue thing

Reply:
Added in 3cb11f738c915058ef7f4d239cdf0da3f1b78f19